authentication/typescript/azure/src/azureIdentityAuthenticationProvider.ts

```diff
@@ -8,8 +8,12 @@ export class AzureIdentityAuthenticationProvider extends BaseBearerTokenAuthenti
     *@param credentials The tokenCredential implementation to use for authentication.
     *@param scopes The scopes to use for authentication.
     *@param options The options to use for authentication.
+    *@param allowedHosts The allowed hosts to use for authentication.
     */
-    public constructor(credentials: TokenCredential, scopes: string[] = ['https://graph.microsoft.com/.default'], options?: GetTokenOptions) {
-        super(new AzureIdentityAccessTokenProvider(credentials, scopes, options));
+    public constructor(credentials: TokenCredential,
+        scopes: string[] = ['https://graph.microsoft.com/.default'],
+        allowedHosts?: string[],
```

Comment:
Pass a set<string> directly. The default hosts set will be empty.

Reply:
```suggestion
        allowedHosts: string[] = [],
```
Do you mean this?

Reply:
allowedHosts?: Set<string>

Reply:
Will we have any default values of hosts and scopes in Kiota?

Reply:
After we handle #1063 in each language, I don't think so.
```suggestion
        allowedHosts: Set<string> = new Set<string>(),
```

so this?

Reply:
Yeah

Reply:
That way we could avoid the array to set conversion.